File: lint-arwaky/src-python/taxonomy/mcp_constants_vo.py
"""MCP-specific constraints and versioning constants."""

# ── Version constants ──────────────────────────────────────────────────────

MCP_SERVER_VERSION = "1.5.0"
MCP_PROTOCOL_MIN = "2024-11-05"
MCP_PROTOCOL_MAX = "2025-06-18"
AUTO_LINT_VERSION = "1.9.3"

# ── Input validation bounds ────────────────────────────────────────────────

MAX_FILES = 1000
MAX_PATH_LENGTH = 1024
MAX_STRING_LENGTH = 8192
MAX_PATH_DEPTH = 50
MAX_BATCH_SIZE = 50

ALLOWED_EXTENSIONS = {
    ".py",
    ".js",
    ".ts",
    ".jsx",
    ".tsx",
    ".mjs",
    ".cjs",
    ".css",
    ".html",
    ".json",
    ".yaml",
    ".yml",
    ".toml",
    ".ini",
    ".cfg",
    ".txt",
    ".md",
    ".rs",
    ".go",
    ".java",
    ".rb",
    ".php",
    ".sh",
    ".bash",
    ".zsh",
    ".dockerfile",
    "",  # allow extensionless files (e.g. Dockerfile, Makefile)
}

File: lint-arwaky/src-rust/taxonomy/mcp_constants_vo.rs
pub const MCP_SERVER_VERSION: &str = "1.5.0";
pub const MCP_PROTOCOL_MIN: &str = "2024-11-05";
pub const MCP_PROTOCOL_MAX: &str = "2025-06-18";
pub const AUTO_LINT_VERSION: &str = "1.9.3";

pub const MAX_FILES: usize = 1000;
pub const MAX_PATH_LENGTH: usize = 1024;
pub const MAX_STRING_LENGTH: usize = 8192;
pub const MAX_PATH_DEPTH: usize = 50;
pub const MAX_BATCH_SIZE: usize = 50;

pub fn allowed_extensions() -> std::collections::HashSet<&'static str> {
    let mut set = std::collections::HashSet::new();
    for ext in &[".py", ".js", ".ts", ".jsx", ".tsx", ".mjs", ".cjs", ".css", ".html",
                  ".json", ".yaml", ".yml", ".toml", ".ini", ".cfg", ".txt", ".md", ".rs",
                  ".go", ".java", ".rb", ".php", ".sh", ".bash", ".zsh", ".dockerfile", ""] {
        set.insert(*ext);
    }
    set
}
